#### aarch64-darwin

```console
connorbaker@Connors-MacBook-Pro nixpkgs % nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"
$ git -c fetch.prune=false fetch --no-tags --force https://github.com/NixOS/nixpkgs master:refs/nixpkgs-review/0
$ git worktree add /Users/connorbaker/.cache/nixpkgs-review/rev-78a439e5143ddf9e1cb0e485ea05e74f61635b40-1/nixpkgs d96a67d15672df117e1ba56aa98c74176d747f66
Preparing worktree (detached HEAD d96a67d1567)
Updating files: 100% (30018/30018), done.
HEAD is now at d96a67d1567 Merge pull request #161545 from Infinisil/synaesthesia
$ nix-env --option system aarch64-darwin -f /Users/connorbaker/.cache/nixpkgs-review/rev-78a439e5143ddf9e1cb0e485ea05e74f61635b40-1/nixpkgs -qaP --xml --out-path --show-trace
$ git merge --no-commit --no-ff 78a439e5143ddf9e1cb0e485ea05e74f61635b40
Automatic merge went well; stopped before committing as requested
$ nix-env --option system aarch64-darwin -f /Users/connorbaker/.cache/nixpkgs-review/rev-78a439e5143ddf9e1cb0e485ea05e74f61635b40-1/nixpkgs -qaP --xml --out-path --show-trace --meta
3 packages updated:
python3.10-rpy2 python3.9-rpy2 R

$ nix --experimental-features nix-command build --no-link --keep-going -f /Users/connorbaker/.cache/nixpkgs-review/rev-78a439e5143ddf9e1cb0e485ea05e74f61635b40-1/build.nix
error: builder for '/nix/store/xy3aiqplz9kl3hafg68bmjjfcx8yh06a-r-nloptr-2.0.0.drv' failed with exit code 1;
       last 10 log lines:
       > /nix/store/qc4fzcja2bs760vsqcsc6v28imjwy0wi-clang-wrapper-11.1.0/bin/cc -I"/nix/store/adk900f75hnpxm8ngrzcx5biqh5mnzq0-R-4.1.3/lib/R/include" -DNDEBUG -I../inst/include -I/nix/store/i5iivfvg4rgfgl0dcw0jra2lgbypwil3-nlopt-2.7.1/include -I'/nix/store/r7inq6x3gv3cf15wwr0q75yq7vapjfvr-r-testthat-3.1.2/library/testthat/include' -isystem /nix/store/4cf902a30yq8z3fgzjr9cax69cyl6qqj-libcxx-11.1.0-dev/include/c++/v1   -fPIC  -g -O2  -c init_nloptr.c -o init_nloptr.o
       > /nix/store/qc4fzcja2bs760vsqcsc6v28imjwy0wi-clang-wrapper-11.1.0/bin/cc -I"/nix/store/adk900f75hnpxm8ngrzcx5biqh5mnzq0-R-4.1.3/lib/R/include" -DNDEBUG -I../inst/include -I/nix/store/i5iivfvg4rgfgl0dcw0jra2lgbypwil3-nlopt-2.7.1/include -I'/nix/store/r7inq6x3gv3cf15wwr0q75yq7vapjfvr-r-testthat-3.1.2/library/testthat/include' -isystem /nix/store/4cf902a30yq8z3fgzjr9cax69cyl6qqj-libcxx-11.1.0-dev/include/c++/v1   -fPIC  -g -O2  -c nloptr.c -o nloptr.o
       > /nix/store/qc4fzcja2bs760vsqcsc6v28imjwy0wi-clang-wrapper-11.1.0/bin/c++ -std=gnu++11 -I"/nix/store/adk900f75hnpxm8ngrzcx5biqh5mnzq0-R-4.1.3/lib/R/include" -DNDEBUG -I../inst/include -I/nix/store/i5iivfvg4rgfgl0dcw0jra2lgbypwil3-nlopt-2.7.1/include -I'/nix/store/r7inq6x3gv3cf15wwr0q75yq7vapjfvr-r-testthat-3.1.2/library/testthat/include' -isystem /nix/store/4cf902a30yq8z3fgzjr9cax69cyl6qqj-libcxx-11.1.0-dev/include/c++/v1   -fPIC  -g -O2  -c test-C-API.cpp -o test-C-API.o
       > /nix/store/qc4fzcja2bs760vsqcsc6v28imjwy0wi-clang-wrapper-11.1.0/bin/c++ -std=gnu++11 -I"/nix/store/adk900f75hnpxm8ngrzcx5biqh5mnzq0-R-4.1.3/lib/R/include" -DNDEBUG -I../inst/include -I/nix/store/i5iivfvg4rgfgl0dcw0jra2lgbypwil3-nlopt-2.7.1/include -I'/nix/store/r7inq6x3gv3cf15wwr0q75yq7vapjfvr-r-testthat-3.1.2/library/testthat/include' -isystem /nix/store/4cf902a30yq8z3fgzjr9cax69cyl6qqj-libcxx-11.1.0-dev/include/c++/v1   -fPIC  -g -O2  -c test-runner.cpp -o test-runner.o
       > /nix/store/qc4fzcja2bs760vsqcsc6v28imjwy0wi-clang-wrapper-11.1.0/bin/c++ -std=gnu++11 -dynamiclib -Wl,-headerpad_max_install_names -undefined dynamic_lookup -single_module -multiply_defined suppress -L/nix/store/adk900f75hnpxm8ngrzcx5biqh5mnzq0-R-4.1.3/lib/R/lib -L/nix/store/44hhgyhbyfikbhjgpjzsg8pq6w6pmjm7-libcxx-11.1.0/lib -o nloptr.so init_nloptr.o nloptr.o test-C-API.o test-runner.o -L/nix/store/0zrx59455j5r39jfgqzfvfbywwsyj3mi-lapack-3/lib -llapack -L/nix/store/irgaxnjgld6bpdrfsjqn4pqdvc6rnqvf-blas-3/lib -lblas -L/nix/store/jk1kxirz192w47xpcd0xbl3swilx55ki-gfortran-11.2.0-lib/lib -L/nix/store/pglwgdvjkpf0913n6d00cqixq2cgmr68-gfortran-wrapper-11.2.0/bin -L/nix/store/d76hvbyddin235hs2in2srqnk1m4qh6k-gfortran-11.2.0/lib/gcc/aarch64-apple-darwin/11.2.0 -L/nix/store/d76hvbyddin235hs2in2srqnk1m4qh6k-gfortran-11.2.0/lib -lintl -liconv -lgfortran -lm -L/nix/store/i5iivfvg4rgfgl0dcw0jra2lgbypwil3-nlopt-2.7.1/lib -lnlopt -L/nix/store/adk900f75hnpxm8ngrzcx5biqh5mnzq0-R-4.1.3/lib/R/lib -lR -lintl -Wl,-framework -Wl,CoreFoundation
       > ld: library not found for -liconv
       > clang-11: error: linker command failed with exit code 1 (use -v to see invocation)
       > make: *** [/nix/store/adk900f75hnpxm8ngrzcx5biqh5mnzq0-R-4.1.3/lib/R/share/make/shlib.mk:10: nloptr.so] Error 1
       > ERROR: compilation failed for package ‘nloptr’
       > * removing ‘/nix/store/9j1r2ry937s9kcr4pg0mbvf4q8p9ly3a-r-nloptr-2.0.0/library/nloptr’
       For full logs, run 'nix log /nix/store/xy3aiqplz9kl3hafg68bmjjfcx8yh06a-r-nloptr-2.0.0.drv'.
error: 1 dependencies of derivation '/nix/store/d3g9v1ysww698y7mcnk98vn0y78fwx4k-r-lme4-1.1-28.drv' failed to build
error: 1 dependencies of derivation '/nix/store/im3dlln4s1w3r0069zxqrqg26x6kvyhb-python3.10-rpy2-3.4.5.drv' failed to build
error: 1 dependencies of derivation '/nix/store/acnb0fmqm9f74dw70x5y8wp259dp17jm-python3.9-rpy2-3.4.5.drv' failed to build
error: 2 dependencies of derivation '/nix/store/ylq80hjsy5kl0ng79jsygds9iffgp7la-review-shell.drv' failed to build
2 packages failed to build:
python310Packages.rpy2 python39Packages.rpy2

1 package built:
rWrapper

error: build log of '/nix/store/im3dlln4s1w3r0069zxqrqg26x6kvyhb-python3.10-rpy2-3.4.5.drv' is not available
error: build log of '/nix/store/539h9p5syf3rw1lfjm0cxdxrpjspgqyg-python3.10-rpy2-3.4.5' is not available
error: build log of '/nix/store/acnb0fmqm9f74dw70x5y8wp259dp17jm-python3.9-rpy2-3.4.5.drv' is not available
error: build log of '/nix/store/yl86ddsd5ri4khswsan0jpijlf9120qi-python3.9-rpy2-3.4.5' is not available
error: build log of '/nix/store/vr4rbfkxps84n3d5m6dpai7nrzbf8jda-R-4.1.3-wrapper.drv' is not available
error: build log of '/nix/store/diaggzv6mf3x03xq8xv8njhi47acnkqg-R-4.1.3-wrapper' is not available
$ nix-shell /Users/connorbaker/.cache/nixpkgs-review/rev-78a439e5143ddf9e1cb0e485ea05e74f61635b40-1/shell.nix

[nix-shell:~/.cache/nixpkgs-review/rev-78a439e5143ddf9e1cb0e485ea05e74f61635b40-1]$ 
```

--- a/pkgs/development/r-modules/default.nix
+++ b/pkgs/development/r-modules/default.nix
@@ -362,7 +362,7 @@ let
     mwaved = [ pkgs.fftw.dev ];
     mzR = with pkgs; [ zlib boost159.dev netcdf ];
     ncdf4 = [ pkgs.netcdf ];
-    nloptr = with pkgs; [ nlopt pkg-config ];
+    nloptr = with pkgs; [ nlopt pkg-config libiconv ];
     n1qn1 = [ pkgs.gfortran ];
     odbc = [ pkgs.unixODBC ];
     pander = with pkgs; [ pandoc which ];

